cloudformation/bootstrap/bootstrap.yaml

```
@@ -0,0 +1,34 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: 'Cirrus Bootstrap - Creates S3 bucket for deployment artifacts'
+
+Parameters:
+  ResourcePrefix:
+    Type: String
+    Description: Prefix for all resource names
+    MinLength: 2
+    MaxLength: 22
+    AllowedPattern: '^[a-zA-Z][a-zA-Z0-9-]*$'
+    ConstraintDescription: Must begin with a letter and contain only alphanumeric characters and hyphens
+    Default: cirrus-sandbox
+
+Resources:
+  CirrusDeploymentArtifactsBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !Sub '${ResourcePrefix}-deployment-artifacts-${AWS::AccountId}'
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: true
+        BlockPublicPolicy: true
+        IgnorePublicAcls: true
+        RestrictPublicBuckets: true
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: AES256
+
+Outputs:
+  CirrusDeploymentArtifactsBucket:
+    Description: S3 bucket for deployment artifacts (Lambda packages and CloudFormation templates)
+    Value: !Ref CirrusDeploymentArtifactsBucket
+    Export:
+      Name: !Sub '${AWS::StackName}-CirrusDeploymentArtifactsBucket'
```

Should the `ResourcePrefix` parameter be programatically carried through to the cirrus deployment via outputs?  Seems like that would reduce the likelihood of typo-induced disconnects.  Or maybe it is OK/advantageous to allow for different prefixes?
```suggestion
  CirrusDeploymentArtifactsBucket:
    Description: S3 bucket for deployment artifacts (Lambda packages and CloudFormation templates)
    Value: !Ref CirrusDeploymentArtifactsBucket
    Export:
      Name: !Sub '${AWS::StackName}-CirrusDeploymentArtifactsBucket'
  CirrusResourcePrefix:
    Description: Prefix to be used by cirrus deployment utilizing this bootstrapping.
    Value: !Ref ResourcePrefix
```

---

I'm more thinking that in the process of deploying the main stack, the ResourcePrefix parameter gets passed via querying the bootstrap stack.  Maybe the illusion of tight coupling.

I personally like separate bootstraps for keeping it data-parallel / avoiding-stepping-on-toes.

---

>I personally like separate bootstraps for keeping it data-parallel / avoiding-stepping-on-toes.

True. I guess pulling it explicitly from the bootstrap stack would guarantee that. But both the bootstrap stack and main stack use the same parameters.json, which contains `ResourcePrefix`. Kinda does the thing, right? I'm afeared that passing that resource prefix between stacks will muddy the waters in terms of what a user might expect the content of the parameters.json to do.